src/v/cloud_storage/partition_manifest.cc

```diff
@@ -521,6 +535,9 @@ bool partition_manifest::add(
         _last_uploaded_compacted_offset = std::max(
           meta.committed_offset, _last_uploaded_compacted_offset);
     }
+
+    _cloud_log_size_bytes += (meta.size_bytes - total_replaced_size);
```

Comment:
This could overflow when we're replacing normal segments with compacted segments, in this case size of the normal segments will always be greater than the size of the compacted segments and `meta.size_bytes` will be less than `total_replaced_size` both of which are `size_t`.

Reply:
Very good point. Funnily enough this works as expected due to modulo arithmetic properties and if well defined behaviour. I'll update the code though.